docs/tutorials/go/audiobook/index.md

```diff
@@ -0,0 +1,1026 @@
+- **`Process`**: Sends a text chunk to OpenAI for processing, retrieves the TTS audio segment, and appends it to an output file.
+- **`MoveOutputFileToPlace`**: Moves an audio file to a safe, versioned location in the same folder as the original text file.
+
+Here is how the activities help in the overall conversion process:
+![After reading a text file and dividing it into chunks, each chunk is sent to OpenAI to be converted to audio and the results appended to the output file](images/highlevelprocess.png)
+
+### Prepare your text with `ReadFile`
+
+The `ReadFile` Activity reads and processes text from a file.
+Before and during the read, it performs a series of safety checks, such as making sure the file exists and it's readable.
+If it encounters any file system issues, it returns an application error.
+A corrupt file system can't be reasonably retried.
+
+The `ReadFile` activity reads and processes text from a file.
+It performs safety checks to ensure the file exists and is readable.
+If file system issues are encountered, it returns an error.
+Errors related to file system issues indicate fundamental issues that can't be resolved through retries.
+
+The following code appears at the end of the Activity.
+It splits the text into "chunks", each containing part of the source material:
+
+```go
+const (
+    apiEndpoint          = "https://api.openai.com/v1/audio/speech"
+    contentType          = "application/json"
+    requestTimeout       = 30 * time.Second
+    maxTokens            = 512
+    averageTokensPerWord = 1.33
+)
+
+...
+
+trimmedContent := strings.TrimSpace(string(content))
+words := strings.Fields(trimmedContent)
+chunks := []string{}
+chunk := strings.Builder{}
+
+for _, word := range words {
+    if float64(chunk.Len()+len(word))*averageTokensPerWord <= maxTokens {
+        if chunk.Len() > 0 {
+            chunk.WriteString(" ")
+        }
+        chunk.WriteString(word)
+    } else {
+        chunks = append(chunks, chunk.String())
+        chunk.Reset()
+        chunk.WriteString(word)
+    }
+}
+
+if chunk.Len() > 0 {
+    chunks = append(chunks, chunk.String())
+}
+
+return chunks, nil
+
+```
+
+As a rule-of-thumb, a typical English word uses 1.33 OpenAI tokens, which is how this code defines the `AVERAGE_TOKENS_PER_WORD` constant.
+When working with other languages, adjust that value to fit with typical word lengths.
+
+Tokens quantify the data processed by OpenAI requests and all OpenAI endpoints use token limits.
+This code creates chunks with approximately 512 token for each API request.
+Although the OpenAI token limit is higher than this, this code is conservative to reduce bandwidth for the returned audio bytes.
+
+After building a method to process string data, you'll add code to build an output file in the next steps.
+
+### Build a temporary file to store output with `CreateTemporaryFile`
+
+The `CreateTemporaryFile` activity requests that the system create a new temporary file.
```

Comment:
By now, I'm confused whether I should be appending these additional functions to an in-progress script, or whether you just wanted me to copy and paste the whole darn thing above and are just walking through it now. Since you've gone to the trouble of walking through it step by step already, I would remove the whole file contents up above and rewrite that part so it's clear you'll be building it incrementally throughout the tutorial.

Reply:
Brian said: build the file first, explain after. I'll make that clearer for each section.